### PR TITLE
systemd template and instructions

### PR DESCRIPTION
**Pull request by bkloosterboer**
### Description

Adds a template and instructions for starting and enabling Impostor using systemd.

### CODE REVIEW

File: docs/Server-configuration.md [hunk 1]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+Linux users will likely want to create a systemd service definition to manage running the Impostor application.  Benefits to using a systemd service include the following:
+
+- The server process can automatically restart if the process dies
+- Impostor can be set to start on boot (handy if you do system updates regularly)
+- You can use systemd targets to ensure the process doesn't start until the system dependencies are met
+
+Using your favorite text editor, create the following service file:
+
+```
+sudo vim /etc/systemd/system/impostor.service
```
Comment (miniduikboot) [inline docs/Server-configuration.md:108]: ooh, getting people stuck in vim, a true classic.

Maybe it's nicer to use `sudoedit` instead? Then it automatically opens in whatever editor they have configured

File: docs/Server-configuration.md [hunk 2]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+```
+
+Modify the following template to your needs and populate the service file.
+
+```
+[Unit]
+Description=Impostor private Among Us server - https://github.com/Impostor/Impostor
```
Comment (miniduikboot) [inline docs/Server-configuration.md:115]: I'm looking at my own unit file (probably should've upstreamed that sooner but oops) and I also have two exta lines here:

```
After=syslog.target
After=network.target
```

This seems to be orthagonal to WantedBy, should I add these lines here or are they sort of obsolete?

File: docs/Server-configuration.md [hunk 3]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+WorkingDirectory=/opt/impostor
+# Path to Impostor binary
+ExecStart=/opt/impostor/Impostor.Server
+Restart=always
+# Restart service after 10 seconds if the dotnet service crashes
+RestartSec=10
+KillSignal=SIGINT
+SyslogIdentifier=log.impostor
+
+# User and group that will run the Impostor server process -- ensure that user has execute on the binary
```
Comment (miniduikboot) [inline docs/Server-configuration.md:128]:
```suggestion
# User and group that will run the Impostor server process -- ensure that user is allowed to execute the binary
```

I think this is more clear to the more casual linux user

File: docs/Server-configuration.md [hunk 4]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+[Service]
+# Directory where Impostor is installed
+WorkingDirectory=/opt/impostor
+# Path to Impostor binary
+ExecStart=/opt/impostor/Impostor.Server
+Restart=always
+# Restart service after 10 seconds if the dotnet service crashes
+RestartSec=10
+KillSignal=SIGINT
+SyslogIdentifier=log.impostor
```
Comment (miniduikboot) [inline docs/Server-configuration.md:126]: Hmm, what is the effect of this line? Can you share an example log line?

For example, this is a log entry on my server:
```
Jan 19 23:15:24 euhtzfsn1 Impostor.Server[11330]: [23:15:24 WRN] Received RpcFlag for unregistered NetId 4.
```

File: docs/Server-configuration.md [hunk 5]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+
+[Service]
+# Directory where Impostor is installed
+WorkingDirectory=/opt/impostor
+# Path to Impostor binary
+ExecStart=/opt/impostor/Impostor.Server
+Restart=always
+# Restart service after 10 seconds if the dotnet service crashes
+RestartSec=10
+KillSignal=SIGINT
```
Comment (miniduikboot) [inline docs/Server-configuration.md:125]: Hmm, I don't have this value specified in my config file. The default KillSignal seems to be SIGTERM, does SIGINT make a difference here? I tested it on my pc and in both cases the client was disconnected gracefully

File: docs/Server-configuration.md [hunk 6]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+WantedBy=multi-user.target
+```
+
+Reload all systemd unit files to source in the new file you've created.
+
+```
+sudo systemctl daemon-reload
+```
+
+Start the service and verify that it starts correctly.  Hint: Use the Q key to quit out of journalctl.
```
Comment (miniduikboot) [inline docs/Server-configuration.md:144]: Hmm, the hint only applies to the second command. Suggestion: add a small explanation after the first command and move the hint to there

File: docs/Server-configuration.md [hunk 7]
```diff
@@ -93,3 +93,66 @@ IMPOSTOR_Server__ListenPort=22023
...
+- Impostor can be set to start on boot (handy if you do system updates regularly)
+- You can use systemd targets to ensure the process doesn't start until the system dependencies are met
+
+Using your favorite text editor, create the following service file:
+
+```
+sudo vim /etc/systemd/system/impostor.service
+```
+
+Modify the following template to your needs and populate the service file.
```
Comment (miniduikboot) [inline docs/Server-configuration.md:111]: Should we put the template in here or should we put it as a separate file that people can just copypaste? (or download from github and use directly)

**Review by miniduikboot — APPROVED** [review]
Thanks for your suggestion. I probably should have added a section like this much earlier.

I've left some remarks, let me know if you agree and I can apply them for you if you want. I'm in favor of merging this PR, but I'd like to merge it as a perfect 10 and not an 6

File: docs/Server-configuration.md (hunk 1 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:108]: that sounds reasonable to me

File: docs/Server-configuration.md (hunk 2 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:115]: WantedBy=multiuser.target adds the multiuser target as a dependency.  I think it's a slightly cleaner way of doing it.  Ultimately your way would also likely work.

File: docs/Server-configuration.md (hunk 3 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:128]: That sounds good to me

File: docs/Server-configuration.md (hunk 4 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:126]: > Jan 20 02:25:39 amongus log.impostor[6741]: [02:25:39 INF] Client 3 disconnecting, reason: The remote sent a disconnect request

File: docs/Server-configuration.md (hunk 5 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:125]: Let's switch it to SIGTERM.  SIGINT sends control + c and after doing some reading, SIGTERM makes more sense.

File: docs/Server-configuration.md (hunk 6 above)
Comment (bkloosterboer) [reply docs/Server-configuration.md:144]: will do

File: docs/Server-configuration.md (hunk 2 above)
Comment (miniduikboot) [reply docs/Server-configuration.md:115]: It's not `After` or `WantedBy`, you want both:
- After specifies this unit must start when a few other units have also started
- WantedBy gives a deadline: this unit must start for the system to consider itself to be fully started.

File: docs/Server-configuration.md (hunk 4 above)
Comment (miniduikboot) [reply docs/Server-configuration.md:126]: ah okay, it just changes the name then. Not sure why we'd need the `log.` prefix then, maybe remove that prefix?

**Comment by miniduikboot** [comment]
Thanks for your PR, I hope this helps a few people with setting up a systemd unit for their servers!

**Pull request merged**